api/ipfsproxy/ipfsproxy.go

```diff
@@ -703,7 +702,7 @@ func (proxy *Server) repoGCHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if !streamErrors && len(errs) != 0 {
-		w.Header().Set("X-Stream-Error", ipfsrepo.NewMultiError(errs...).Error())
+		w.Header().Set("X-Stream-Error", newMultiError(errs...).Error())
```

Comment:
Instead of building a slice of errors, which you then loop to generate a string, build the final error string directly as you receive the errors. Note how the last error is also becoming the "summary" even though it is not a summary.